types.nix

```diff
@@ -57,6 +57,8 @@ rec {
         "${dev}-part${toString index}" # /dev/disk/by-id/xxx style
       else if match "/dev/(nvme|md/|mmcblk).+" dev != null then
         "${dev}p${toString index}" # /dev/nvme0n1p1 style
+      else if match "\\\$\\{.+}" dev != null then # shell variable: ${primaryDisk}
+        "${dev}${toString index}"
```

Comment:
well this also just works if you give it a "/dev/xda" style device. maybe we need some more logic here if people want to use this with nvmes etc.

Reply:
Oh right! Not entirely sure how to solve that tbh, as we can't know which scheme to use during eval time. 

The only idea I've got for the moment would be to move device numbering to run-time (at least in this case) but that seems somewhat more involved. :/ Happy for any input here :)

Reply:
after reading the old code, it actually did somewhat different, but we can reuse that here also. It tried to find the corresponding device in either /dev/disk/by-id or /dev/disk/by-path and continued to work with that (since the device numbering is stable with those).
https://github.com/nix-community/disko/blob/9bb4aec9640cbc30e241c267158e506278862b5e/default.nix#L7